test/unit_tests/test_eegneuralnet.py

```diff
@@ -0,0 +1,353 @@
+    # for cropped decoding classifier returns one label for each trial (averaged over all crops)
+    np.testing.assert_array_equal(target_predict, eegneuralnet.predict(MockDataset()))
+    # for cropped decoding classifier returns values for each trial (average over all crops)
+    np.testing.assert_array_equal(preds.mean(-1), eegneuralnet.predict_proba(MockDataset()))
+
+
+def test_cropped_predict_and_predict_proba_not_aggregate_predictions(eegneuralnet_cls, preds):
+    eegneuralnet = eegneuralnet_cls(
+        MockModuleReturnMockedPreds,
+        module__preds=preds,
+        module__n_outputs=4,
+        module__n_chans=3,
+        module__n_times=3,
+        cropped=True,
+        criterion=CroppedLoss,
+        criterion__loss_function=nll_loss,
+        optimizer=optim.Adam,
+        batch_size=32,
+        aggregate_predictions=False
+    )
+    eegneuralnet.initialize()
+    target_predict = preds if isinstance(eegneuralnet, EEGRegressor) else preds.argmax(1)
+    np.testing.assert_array_equal(target_predict, eegneuralnet.predict(MockDataset()))
+    np.testing.assert_array_equal(preds, eegneuralnet.predict_proba(MockDataset()))
+
+
+def test_predict_trials(eegneuralnet_cls, preds):
+    eegneuralnet = eegneuralnet_cls(
+        MockModuleReturnMockedPreds,
+        module__preds=preds,
+        module__n_outputs=4,
+        module__n_chans=3,
+        module__n_times=3,
+        cropped=False,
+        criterion=CroppedLoss,
+        criterion__loss_function=nll_loss,
+        optimizer=optim.Adam,
+        batch_size=32
+    )
+    eegneuralnet.initialize()
+    with pytest.warns(UserWarning, match="This method was designed to predict "
+                                         "trials in cropped mode."):
+        eegneuralnet.predict_trials(MockDataset(), return_targets=False)
+
+
+def test_clonable(eegneuralnet_cls, preds):
+    eegneuralnet = eegneuralnet_cls(
+        MockModuleReturnMockedPreds,
+        module__preds=preds,
+        module__n_outputs=4,
+        module__n_chans=3,
+        module__n_times=3,
+        cropped=False,
+        callbacks=[
+            "accuracy",
+            ("lr_scheduler", LRScheduler('CosineAnnealingLR', T_max=1))],
+        criterion=CroppedLoss,
+        criterion__loss_function=nll_loss,
+        optimizer=optim.Adam,
+        batch_size=32
+    )
+    clone(eegneuralnet)
+    eegneuralnet.initialize()
+    clone(eegneuralnet)
+
+
+def test_set_signal_params_numpy(eegneuralnet_cls, preds, Xy):
+    X, y = Xy
+    net = eegneuralnet_cls(
+        MockModuleFinalLayer,
+        module__preds=preds,
+        cropped=False,
+        optimizer=optim.Adam,
+        batch_size=32,
+        train_split=None,
+        max_epochs=1,
+    )
+    net.fit(X, y=y)
+    assert net.module_.n_times == 10
+    assert net.module_.n_chans == 3
+    assert net.module_.n_outputs == (1 if isinstance(net, EEGRegressor) else 4)
+
+
+def test_set_signal_params_epochs(eegneuralnet_cls, preds):
```

Comment:
just a pass?

Reply:
sorry it was not meant to be committed. This test will be for #529